client/components/button/button.mjs

```diff
@@ -1,6 +1,6 @@
 /* Exported, publicly available function, use wherever needed in code */
 
 export function useButton() {
-  const button = document.getElementById('button');
+  const button = document.getElementById('buttonPrimary');
   button.onclick = () => console.log('clicked');
-}
+}
```

Comment:
Oh, great problem you have here.  
So this line:  
```
button.onclick = () => console.log('clicked');
```
is what makes your function hardcoded and not reusable at all.
Now, let's think what we want: 
1. We want other developers to use this function **without rewriting it**
2. We want other developers to define what happens `onclick`: by putting console.log we assume every user of this function wants only this functionality. Which is incorrect, of course.

Can you figure out how? I can help also, but try to tackle this one yourself.

Reply:
Thank you for your comment, but could you clarify my task for JS functionality, please?
I've created template of two type of buttons (Primary and Ghost) in HTML and CSS files. I thought, javascript functionality will be created during coding of main block. Because, same type of button will have different functions in different places of application, isn't it? Or should I think about all variants of using each of buttons in everywhere in App and create templates for every function?

Reply:
If this wasn't part of the task, then you shouldn't include it in pull request. Cause reviewer then thinks it was ;)

These are criteria you can follow:
- We want other developers to use this function without rewriting it
- We want other developers to define what happens onclick

The change is actually fairly simple. And, indeed, other developers will be defining what happens onclick, but this small piece of JS: finding button by id and assigning onclick to it would be repeated by **everyone**.
Hence you could provide this useButton as reusable function, which accepts button id and callback to execute.

Reply:
I hope, I've done it.
Please, keep in mind, that file index.mjs is file for checking how dose it work.
Work file of JS is button.js

Reply:
No it's still not reusable. We don't want developers to copy, but reuse this function. This is possible because of JS modules we are making. So here is how to make it work:
```
/**
@id id of button
@onclick callback function to be executed on button click
**/
export function useButton(id, onClick) {
  const button = document.getElementById(id);
  button.onclick = onClick;
}
```

It's fairly simple, but may spare some repetitive code.
Now, others could use it like this:
HTML:
```
<button id="login">Login</button>
```
JS:
```
import {useButton} from '../path/to/button.mjs';


function openLoginModal() {
  // ... here some code to open some modal window
}

// Here you avoid hassle of searching for button and specifying onclick
useButton('login', openLoginModal);
```

Reply:
Sorry, but I still can't understand, what is the difference in your variant in compare with mine.

My variant:
**Button.mjs:**
<img width="579" alt="image" src="https://user-images.githubusercontent.com/108457205/199722268-776007b3-cda6-40f3-9723-13611bd95eea.png">

And test of this solution in file **index.mjs:**
<img width="411" alt="image" src="https://user-images.githubusercontent.com/108457205/199722552-d18bce87-450b-4c95-8657-cc9c7a3411b8.png">

Please, explain me, where I am wrong.
Thanks.

Reply:
No all correct, actually. I didn't see button.mjs file when was checking, saw only example code